src/structlog/processors.py

```diff
@@ -531,3 +543,184 @@ def __call__(
+        ),
+        CallsiteParameter.THREAD: (
+            lambda module, frame_info: threading.get_ident()
+        ),
+        CallsiteParameter.THREAD_NAME: (
+            lambda module, frame_info: threading.current_thread().name
+        ),
+        CallsiteParameter.PROCESS: (lambda module, frame_info: os.getpid()),
+        CallsiteParameter.PROCESS_NAME: (
+            lambda module, frame_info: get_processname()
+        ),
+    }
+    _record_attribute_map: ClassVar[Dict[CallsiteParameter, str]] = {
+        CallsiteParameter.PATHNAME: "pathname",
+        CallsiteParameter.FILENAME: "filename",
+        CallsiteParameter.MODULE: "module",
+        CallsiteParameter.FUNC_NAME: "funcName",
+        CallsiteParameter.LINENO: "lineno",
+        CallsiteParameter.THREAD: "thread",
+        CallsiteParameter.THREAD_NAME: "threadName",
+        CallsiteParameter.PROCESS: "process",
+        CallsiteParameter.PROCESS_NAME: "processName",
+    }
+
+    _all_parameters: Set[CallsiteParameter] = set(CallsiteParameter)
+
+    class _RecordMapping(NamedTuple):
+        event_dict_key: str
+        record_attribute: str
+
+    __slots__ = [
+        "_active_handlers",
+        "_additional_ignores",
+        "_record_mappings",
+    ]
+
+    def __init__(
+        self,
+        parameters: Collection[CallsiteParameter] = _all_parameters,
+        additional_ignores: Optional[List[str]] = None,
+    ) -> None:
+        if additional_ignores is None:
+            additional_ignores = []
+        # Ignore stack frames from the logging module. They will occur if this
+        # processor is used in ProcessorFormatter, and additionally the logging
+        # module should not be logging using structlog.
+        self._additional_ignores = ["logging", *additional_ignores]
+        self._active_handlers: List[
+            Tuple[CallsiteParameter, Callable[[str, inspect.Traceback], Any]]
+        ] = []
+        self._record_mappings: List[
+            "CallsiteParameterAdder._RecordMapping"
+        ] = []
+        for parameter in parameters:
+            self._active_handlers.append(
+                (parameter, self._handlers[parameter])
+            )
+            self._record_mappings.append(
+                self._RecordMapping(
+                    parameter.value,
+                    self._record_attribute_map[parameter],
+                )
+            )
+
+    def __call__(
+        self, logger: logging.Logger, name: str, event_dict: EventDict
+    ) -> EventDict:
+        record: Optional[logging.LogRecord] = event_dict.get("_record")
+        from_structlog: Optional[bool] = event_dict.get("_from_structlog")
+        # If the event dictionary has a record, but it comes from structlog,
+        # then the callsite parameters of the record will not be correct.
+        if record is not None and not from_structlog:
+            for mapping in self._record_mappings:
+                event_dict[mapping.event_dict_key] = record.__dict__[
```

Comment:
shouldn't `record[` work here?

Reply:
`record` is a [`logging.LogRecord`](https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L274) and as far as I know this has no features to make that work and it fails in test with `TypeError: 'LogRecord' object is not subscriptable`. It could be `getattr` instead of `__dict__` though and I'm happy to change to `getattr` if you want, may be a bit better.

Reply:
Ah yeah naw it's fine.